src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -305,3 +309,25 @@ WaveformMarkPointer WaveformWidgetRenderer::getCueMarkAtPoint(QPoint point) cons
     }
     return nullptr;
 }
+
+// TODO(hacksdump): Optimize this search
+Beat WaveformWidgetRenderer::getBeatAtPoint(QPoint point) const {
+    for (const auto& pBeat : m_beatPositions.keys()) {
+        float position = m_beatPositions[pBeat];
+        if (getOrientation() == Qt::Horizontal) {
+            if (position - 5 < point.x() && point.x() < position + 5) {
```

Comment:
What is this magic number 5? I presume it's a padding width on either side of the line to make a wider area to right click? Please move it to a named constant in an anonymous namespace.

Reply:
That certainly should be a constant
But I would like to point out a design inconsistency that has been introduced. If you look at the function just above this, it happens to be `getCueMarkAtPoint` which returns a `WaveformMarkPointer`, which is the graphic object for the mark.
But the return type of current function is not WaveformBeat / WaveformBeatPointer but the Beat object itself.
This means I can't currently use the `contains` method used with WaveformMarkPointer to check if point is in vicinity.
To actually start using WaveformBeat insead of Beat, I would need to store the beat object inside the WaveformBeat class.

A positive side for making this consistent would be:
We can create an interface like `WaveformMarkRightClickable` (for the lack of a good name) which has a method `contains` and `WaveformMark` and `WaveformBeat` both have to implement that interface. I mean pattern comes from Java, I'd like to hear your suggestions on how to do it in C++.

Reply:
That seems like a good idea. Then we could have less special cases to handle in WWaveformWidgetRenderer.